app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -484,15 +485,27 @@ class PluginApi {
 
       if (siteSettings.enable_experimental_sidebar_hamburger) {
         try {
-          const { route, label, rawLabel, className } = fn();
+          const { href, route, label, rawLabel, className } = fn();
           const textContent = rawLabel || I18n.t(label);
 
-          this.addCommunitySectionLink({
+          const args = {
             name: className || textContent.replace(/\s+/g, "-").toLowerCase(),
-            route,
             title: textContent,
             text: textContent,
-          });
+          };
+
+          if (href) {
+            if (DiscourseURL.isInternal(href)) {
+              args.href = href;
+            } else {
+              // Skip external links support for now
+              return;
+            }
+          } else {
+            args.route = route;
+          }
+
+          this.addCommunitySectionLink(args);
         } catch {
```

Comment:
I know this was from yesterday's PR, but should we change this to `finally`? It's possible that this `try` block will never throw an error (this the scenario that we hope for), but I do think that we should log the deprecation anyway because we aren't going to keep this bridge forever, right?

Reply:
Right now I plan to maintain the bridge until `enable_experimental_sidebar_hamburger` is removed because that is when the new sidebar is enabled by default for everyone. When the `enable_experimental_sidebar_hamburger` flag is removed, I will then log the deprecation regardless. Right now, some usage of the `decorateWidget` API is impossible to build a bridge API for because it relies on certain widget states.